src/Symfony/Component/FeatureFlag/Provider/ChainProvider.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag\Provider;
+
+/**
+ * @experimental
+ */
+final class ChainProvider implements ProviderInterface
+{
+    public function __construct(
+        /** @var list<ProviderInterface> */
+        private readonly iterable $providers = [],
```

Comment:
There is no need to maintain code compatibility with PHP 8.1.

```suggestion
final readonly class ChainProvider implements ProviderInterface
{
    public function __construct(
        /** @var list<ProviderInterface> */
        private iterable $providers = [],
```

Reply:
Now that the PR target is 8.0, it's OK for me.